### DISCUSSION TITLE
Is next-server using deprecated `UrlWithParsedQuery` type?

### DISCUSSION THREAD

**Discussion by nbouvrette** [vercel/next.js#21283] [Help] [answered]
I'm using a [custom server](https://nextjs.org/docs/advanced-features/custom-server) with TypeScript and having a hard time passing the right parameters to `getRequestHandler`

The 3rd parameter of this method is expecting `parsedUrl?: UrlWithParsedQuery` and unless I am missing something this type is coming from `import { parse } from 'url';` which has been [deprecated since Node.js 11](https://nodejs.org/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost).

I also see [one of the examples](https://github.com/vercel/next.js/blob/canary/examples/custom-server/server.js) that seems to be using the [WHATWG URL API](https://nodejs.org/api/url.html#url_the_whatwg_url_api) so I am wondering if this is simply just a type issue on next-server?

**Comment by nbouvrette** [comment, marked as answer]
Nevermind, found the answer myself by looking at the original Typescript method:

https://github.com/vercel/next.js/blob/ce5d9c858b70f647b532ec2f62ddde506af32b6a/packages/next/next-server/server/next-server.ts#L291

Looks like this field is redundant. It might be good to remove those deprecated methods from the Next.js examples (and eventually Next.js).

**Comment by PlopTheReal** [comment]
Hey, for some reasons, following the [server example there](https://github.com/vercel/next.js/blob/canary/examples/custom-server/server.js), when I change from 'parse()' to the WHATWG URL APIs, I'm receiving an error on urls that aren't pointing to any page. Using 'parse()' it redirects to index.js. Any idea?

_error returned:_ 
```
TypeError: Cannot read property '__nextLocale' of undefined
    at Router.execute (.../node_modules/next/dist/next-server/server/router.js:17:712)
```

**Reply by nbouvrette** [reply]
@PlopTheReal not sure what you mean? the example is already using the WHATWG URL API. What I meant by my answer is that the `query` variable is redundant and should be removed.

**Reply by PlopTheReal** [reply]
Sorry if my message is a bit confusing. 

When accessing a page that doesn't exists on the server such as `http://host/pageThatDoesntExist`, the following code fails with the error `TypeError: Cannot read property '__nextLocale' of undefined`
```
const parsedUrl = new URL(req.url, 'http://w.w')
handle(req, res, parsedUrl)
```

On the other hand with the following code: 
```
const parsedUrl = parse(req.url, true)
handle(req, res, parsedUrl)
```
Nextjs is able to return index.js which is what I'm expecting.
I'm on nextjs 10.0.5.
